src/tests.c

```diff
@@ -3690,37 +3690,77 @@ void run_wnaf(void) {
     CHECK(secp256k1_scalar_is_zero(&n));
 }
 
+void test_ecmult_accumulate(secp256k1_sha256* acc, const secp256k1_scalar* x) {
+    /* Compute x*G in 4 different ways, serialize it uncompressed, and feed it into acc. */
+    secp256k1_gej rj1, rj2, rj3, rj4, gj, infj;
+    secp256k1_ge r;
+    const secp256k1_scalar zero = SECP256K1_SCALAR_CONST(0, 0, 0, 0, 0, 0, 0, 0);
+    unsigned char bytes[65];
+    size_t size = 65;
+    secp256k1_gej_set_ge(&gj, &secp256k1_ge_const_g);
+    secp256k1_gej_set_infinity(&infj);
+    secp256k1_ecmult_gen(&ctx->ecmult_gen_ctx, &rj1, x);
+    secp256k1_ecmult(&ctx->ecmult_ctx, &rj2, &gj, x, &zero);
+    secp256k1_ecmult(&ctx->ecmult_ctx, &rj3, &infj, &zero, x);
+    secp256k1_ecmult_const(&rj4, &secp256k1_ge_const_g, x, 256);
+    secp256k1_ge_set_gej_var(&r, &rj1);
+    ge_equals_gej(&r, &rj2);
+    ge_equals_gej(&r, &rj3);
+    ge_equals_gej(&r, &rj4);
+    if (secp256k1_ge_is_infinity(&r)) {
+        /* Store infinity as 0x00 */
+        const unsigned char zerobyte[1] = {0};
+        secp256k1_sha256_write(acc, zerobyte, 1);
+    } else {
+        /* Store other points using their uncompressed serialization. */
+        secp256k1_eckey_pubkey_serialize(&r, bytes, &size, 0);
+        CHECK(size == 65);
+        secp256k1_sha256_write(acc, bytes, size);
+    }
+}
+
 void test_ecmult_constants(void) {
-    /* Test ecmult_gen() for [0..36) and [order-36..0). */
+    /* Test ecmult_gen for:
+     * - Numbers 0..36 and their negations
+     * - Numbers 2^i (with i=0..255)
+     * - Numbers 2^i + 2^j (with i=0..255, j=i+1..255)
```

Comment:
`i` only goes up to 254.

Reply:
Well yeah it's equivalent in the end.

Reply:
If it's equivalent I'd prefer `i=0..42069`. Fwiw I'm only mentioning this because I added some printfs to this code and was confused for a moment why there was no `i=255`.

@sipa care to fix the micronit? Fine for me either way.

Reply:
Oh, I was considering an alternative approach to instead test all numbers of the form `a*(2^N)^b a=1..M b=0..256/M`, for some M and N, as that'd give better coverage of the actual tables involved.

Reply:
I've rewritten it to use (j*2^i) for i=0..255 and odd j=1..255.